CRM/Core/DAO.php

```diff
@@ -812,6 +816,11 @@ public function save($hook = TRUE) {
       }
 
       $result = $this->update();
+      if (!empty($this->_custom)) {
+        // @fixme: use a function with API4 syntax
+        // Add customFields back to $result
+        CRM_Core_BAO_CustomValueTable::store($this->_custom, $this->tableName(), $idField, $op);
```

Comment:
This needs to pass the actual id of the main entity's record instead of hte name of the id field